include/zephyr/net/wifi_mgmt.h

```diff
@@ -143,6 +143,10 @@ enum net_event_wifi_cmd {
 
 #define NET_EVENT_WIFI_DISCONNECT_COMPLETE			\
 	(_NET_WIFI_EVENT | NET_EVENT_WIFI_CMD_DISCONNECT_COMPLETE)
+
+struct wifi_scan_params {
+	enum wifi_scan_type scan_type;
```

Comment:
Shouldn't this be just a `bool` (since the parameter signifies whether a scan should be forced to passive or not). Also `scan_type` can be renamed to `forced_passive` to convey the above.

Reply:
Without scan_type, `forced_passive` doesn't make much sense, so, to introduce `scan_type` and then forcing it to passive at wifi_mgmt seems to be a better approach. Also, adding this structure makes way for extending scan API.